docs/GettingStarted.md

```diff
@@ -0,0 +1,80 @@
+
+## Building and running Hail
+
+Hail may be built to run locally with all cores on a single machine or to run on a Spark cluster. Running locally is useful for getting started, analyzing or experimenting with small datasets, and Hail development.
+
+### Building and running Hail locally
+The single command
+```
+./gradlew installDist
+```
+will populate `build/install` with an installation of Hail, including the executible `build/install/hail/bin/hail`. The initial build takes time as the build tool [Gradle](https://gradle.org/) installs and then downloads all Hail build dependencies, including the Scala compiler and Spark.
+
+Here are a few simple things to try. To list all commands, run
+```
+./build/install/hail/bin/hail -h
+```
+To convert the included `sample.vcf` to Hail's `.vds` format, run
+```
+./build/install/hail/bin/hail importvcf src/test/resources/sample.vcf write -o ~/sample.vds
+```
+Then to count the number of samples and variants, run
+```
+./build/install/hail/bin/hail read -i ~/sample.vds count
+```
+To compute and output sample and variant quality control statistics, run
+```
+./build/install/hail/bin/hail read -i ~/sample.vds variantqc -o ~/variantqc.tsv sampleqc -o ~/sampleqc.tsv
+```
+To run Hail using `hail` in place of `./build/install/hail/bin/hail`, add this `bin` directory to your path. Note that during each run Hail writes a `hail.log` file in the current directory; this is useful to developers for debugging.
+
+### Building a Spark application JAR
+
+Hail is compatible with Spark versions 1.5 and 1.6, and uses 1.5.2 by default. Set the Spark version to match that of the Spark cluster with
+```
+./gradlew -Dspark.version=VERSION
+```
+Then run
+```
+./gradlew shadowJar
+```
+to build `build/libs/hail-all-spark.jar`, which can be submitted using `spark-submit` (see [Spark documentation](http://spark.apache.org/docs/1.6.2/submitting-applications.html) for details).
+
+### BLAS and LAPACK
+
+Hail uses the Scala numerical processing library [Breeze](http://github.com/scalanlp/breeze), which in turn uses the [netlib-java](http://github.com/fommil/netlib-java) to take advantage of native [BLAS](http://www.netlib.org/blas/) and [LAPACK](http://www.netlib.org/lapack/) routines. These routines are available by default on OS X and Windows. On Linux, they must be explicitly enabled. On Ubuntu 14.04, run
+```
+apt-get install libatlas-base-dev
+```
+If natives are not found, `hail.log` will contain the warnings
+```
+Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
+Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
+```
+
+## Rebuilding and testing Hail
+
+Subsequent builds may at times fail due to artifacts from previous builds. To delete the `build` directory and start fresh, run
```

Comment:
This is not strictly correct.  Clean does not delete artifacts.  I'd probably leave this out.  This should go in something like tips/tricks in the developer docs.